app/services/company.js

```diff
@@ -3,30 +3,93 @@ const path = require('path');
+const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
 const fileService = require('./file');
 const error = require('../services/error');
 const CrudService = require('./crudService');
 const companyFilesFolder = require('../../config.js').companyFilesFolder;
+const ORIGIN = require('../../config.js').ORIGIN;
 
 class CompanyService extends CrudService {
+  async get(id) {
+    let found = await companyRepository.get(id);
+    if (!found) {
+      error.throwNotFoundError('Company not found.');
+    }
+
+    const { fieldOfActivityId, ...company } = found;
+    const fieldOfActivity =
+      await fieldOfActivityRepository.get(fieldOfActivityId);
+    company.business = fieldOfActivity.name;
+
+    const companyAddresses = await addressRepository.getAllByProps({
+      companyId: id,
+    });
+    if (companyAddresses && companyAddresses.length) {
+      const {
+        streetName,
+        streetNumber,
+        apartmentNumber,
+        city,
+        country,
+      } = companyAddresses[0];
+      company.address = `${streetName} ${streetNumber}, ap. ${apartmentNumber}, ${city}, ${country}`;
+    }
+
+    const companyFiles = await fileRepository.getAllByProps({
+      entityId: id,
+      entityType: 'company',
+    });
+    company.images = companyFiles.map(file => ({
+      name: file.originalFileName,
+      url: `${ORIGIN}companies/${file.originalFileName}`,
+    }));
+
+    return company;
+  }
+
   async create(company, files) {
     if (company.id !== undefined) {
       error.throwValidationError('Invalid company format.');
     }
 
-    const companyExists = await companyRepository.getByPropsNonParanoid({ name: company.name });
+    const companyExists = await companyRepository.getByPropsNonParanoid({
+      name: company.name,
+    });
     if (companyExists) {
       error.throwValidationError('Company already exists.');
     }
-
-    if (company.fieldOfActivityId !== undefined) {
-      const fieldExists = await fieldOfActivityService.get(company.fieldOfActivityId);
+    const { business, address, ...newCompany } = company;
+    if (business !== undefined) {
+      const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({
+        name: business,
+      });
+      if (!fieldOfActivity) {
+        error.throwNotFoundError('Field of activity not found.');
+      }
+      newCompany.fieldOfActivityId = fieldOfActivity.id;
     } else {
       error.throwValidationError('Invalid company format.');
     }
 
-    const createdCompany = await companyRepository.create(company);
+    const createdCompany = await companyRepository.create(newCompany);
+
+    if (address !== undefined) {
+      const addressFields = address.split(', ');
+      const [ streetName, streetNumber ] = addressFields[0].split(' ');
+      const apartmentNumber = addressFields[1].split(' ')[1];
+      const city = addressFields[2];
+      const country = addressFields[3];
+
+      await addressRepository.create({
```

Comment:
Please create a service for the Address if there isn't one already and call it here.